src/lib/consensus/proof_of_stake.ml

```diff
@@ -2264,15 +2371,19 @@ let select ~existing ~candidate ~logger =
         , "candidate last epoch is longer than existing last epoch" ) )
       (* these two could be condensed into one entry *)
     ; ( ( lazy
-            ( existing.curr_epoch_data.lock_checkpoint
-            = candidate.last_epoch_data.lock_checkpoint )
+            (Option.fold existing.curr_epoch_data.lock_checkpoint ~init:false
```

Comment:
You could alternatively write this as `[%derive.eq: State_hash.t option] existing_curr_lock_checkpoint (Some (candidate.last_epoch_data.lock_checkpoint))`. This is method is fine though if you prefer it.